docs/guides/transitions.md

```diff
@@ -323,6 +323,12 @@ Just like transitions, transient transitions can be specified as a single transi
 
 Null events are always "sent" for every transition, internal or external.
 
+::: warning
+
+Transient transitions will be run for every event while the statechart is in the state on which they are defined, so it is important that they use a `target` value instead of simply raising events to avoid creating infinite event loops.
```

Comment:
Hm, the issue itself is not focused on using target over events, but rather about not sending events which are in turn caught (handled) by the state which has that transient transition defined on it - because that cause that problem of transient transitions being checked again.